From 4eb1f9a745a1194df792d124ff5c050befb8eff7 Mon Sep 17 00:00:00 2001
From: Xiao Chen <xiao@cloudera.com>
Date: Thu, 7 Sep 2017 23:19:11 -0700
Subject: [PATCH 2631/2639] Revert "CLOUDERA-BUILD. CDH-57314. Increase the
 default number of retries in KMS clients."

This reverts commit 2714ae1f2eeb44ca49eef758069c925a72acebee.

Change-Id: I6f3751e8c9e061748f3a2c2606a7797aa316a679
(cherry picked from commit 9e0a590977c71a96ebd432852a367604774daa2f)
---
 .../key/kms/LoadBalancingKMSClientProvider.java    |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/LoadBalancingKMSClientProvider.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/LoadBalancingKMSClientProvider.java
index 81b0e32..896b7a9 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/LoadBalancingKMSClientProvider.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/LoadBalancingKMSClientProvider.java
@@ -148,9 +148,9 @@ public LoadBalancingKMSClientProvider(KMSClientProvider[] providers,
               + " providers(depending on {}={} setting and numProviders={})"
               + " in the group OR the exception is not recoverable",
               CommonConfigurationKeysPublic.KMS_CLIENT_FAILOVER_MAX_RETRIES_KEY,
-              getConf().getInt(CommonConfigurationKeysPublic.
-                      KMS_CLIENT_FAILOVER_MAX_RETRIES_KEY,
-                  CommonConfigurationKeysPublic.KMS_CLIENT_FAILOVER_MAX_RETRIES_DEFAULT),
+              getConf().getInt(
+                  CommonConfigurationKeysPublic.
+                  KMS_CLIENT_FAILOVER_MAX_RETRIES_KEY, providers.length),
               providers.length);
           throw ex;
         }
-- 
1.7.9.5

